collect_app/src/test/java/org/odk/collect/android/external/InstanceProviderTest.java

```diff
@@ -80,6 +82,19 @@ public void insert_addsInstance() {
         }
     }
 
+    @Test
```

Comment:
In other tests where we verify exceptions we do that with `@Test(expected = SecurityException.class)`
and then 
```java
try {
    contentResolver.insert(getUri(firstProjectId), values);
    fail();
} catch (SecurityException e) {
    // Expected
}
```
won't be needed you can just call `contentResolver.insert(getUri(firstProjectId), values);`

Reply:
I prefer this approach as it's more specific about where the exception occurs. I generally only use `@Test(expected = SecurityException.class)` for single statement tests.